File: continuum-propel/src/frontends/structures-admin/pages/structures/traits/Trait.ts

export class Trait {

    public id: string
    public name: string
    public describeTrait: string
    public schema: string
    public esSchema: string

    public created: number
    public updated: number

    public required: boolean // should the GUI require a field to be filled out when looking at the item
    public includeInLabel: boolean
    public includeInQRCode: boolean
    public modifiable: boolean // should this field be modifiable outside the system
    public unique: boolean // should be a unique field in the index, so no others should exist
    public operational: boolean // field that says we do not really add to the schema but provide some type of process

    constructor(id: string,
                name: string,
                describeTrait: string,
                schema: string,
                esSchema: string,
                created: number,
                updated: number,
                required: boolean,
                includeInLabel: boolean,
                includeInQRCode: boolean,
                modifiable: boolean,
                unique: boolean,
                operational: boolean) {
        this.id = id
        this.name = name
        this.describeTrait = describeTrait
        this.schema = schema
        this.esSchema = esSchema
        this.created = created
        this.updated = updated
        this.required = required
        this.includeInLabel = includeInLabel
        this.includeInQRCode = includeInQRCode
        this.modifiable = modifiable
        this.unique = unique
        this.operational = operational
    }

}

File: structures/src/main/java/org/kinotic/structures/api/domain/Trait.java
/*
 *
 * Copyright 2008-2021 Kinotic and the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kinotic.structures.api.domain;

import java.io.Serializable;

public class Trait implements Serializable {

    private String id = null;
    private String name = null;
    private String describeTrait = null;
    private String schema = null;
    private String esSchema = null;
    private long created = 0;
    private boolean required = false; // should the GUI require a field to be filled out when looking at the item

    private Long updated;
    private boolean modifiable = true; // should this field be modifiable outside the system
    private boolean unique = false; // should be a unique field in the index, so no others should exist
    private boolean operational = false; // field that says we do not really add to the schema but provide some type of process

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescribeTrait() {
        return describeTrait;
    }

    public void setDescribeTrait(String describeTrait) {
        this.describeTrait = describeTrait;
    }

    public String getSchema() {
        return schema;
    }

    public void setSchema(String schema) {
        this.schema = schema;
    }

    public String getEsSchema() {
        return esSchema;
    }

    public void setEsSchema(String esSchema) {
        this.esSchema = esSchema;
    }

    public long getCreated() {
        return created;
    }

    public void setCreated(long created) {
        this.created = created;
    }

    public boolean isRequired() {
        return required;
    }

    public void setRequired(boolean required) {
        this.required = required;
    }

    public Long getUpdated() {
        return updated;
    }

    public void setUpdated(Long updated) {
        this.updated = updated;
    }

    public boolean isModifiable() {
        return modifiable;
    }

    public void setModifiable(boolean modifiable) {
        this.modifiable = modifiable;
    }

    public boolean isUnique() {
        return unique;
    }

    public void setUnique(boolean unique) {
        this.unique = unique;
    }

    public boolean isOperational() {
        return operational;
    }

    public void setOperational(boolean operational) {
        this.operational = operational;
    }
}
